nessie/src/main/java/org/apache/iceberg/nessie/NessieIcebergClient.java

```diff
@@ -165,10 +166,10 @@ private TableIdentifier toIdentifier(EntriesResponse.Entry entry) {
     return TableIdentifier.of(elements.toArray(new String[elements.size()]));
   }
 
-  public IcebergTable table(TableIdentifier tableIdentifier) {
+  public IcebergTable table(TableIdentifier tableIdentifier, Reference ref) {
```

Comment:
This signature change with all the related changes makes the call sides more complicated - hard to see a justification for that.

Reply:
As mentioned, I changed `renameTable` and `dropTable` (it was not using TableOperation) to be similar to `commitTable`. 

If I don't pass the reference, calling `getRef()` may give refreshed reference (maybe refreshed by concurrent commit).  So, to maintain the same state of reference for a whole operation, I am holding the state and passing it.

Reply:
What's the real world scenario then wrt user interaction?